### PR TITLE
60 individual listing pages

### LINKED ISSUE #60: Individual listing pages

### PR DESCRIPTION
**Pull request by kfoxIsProgrammer**
Added the single listing
Adding to Cart is now moved to the single listing page

Chrome
captures_chrome-capture-2023-3-4

### CODE REVIEW

File: frontend/src/pages/Listing/Listing.tsx [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+  const { sendRequest } = useHttpClient();
+  useEffect(() => {
+    const getSellerItems = async () => {
+      let item: IListingCardProps = await sendRequest(
+        LISTING,
+        "GET",
+        {},
+        `/${listingId}`
+      );
+      setCurrentListing(item);
```
Comment (NosBos) [inline frontend/src/pages/Listing/Listing.tsx:47]: Calling setState inside useEffect is causing an infinite loop of updating. We need to add a dependency array

File: frontend/src/pages/OAuth2RedirectHandler/OAuth2RedirectHandler.tsx [hunk 2]
```diff
@@ -35,10 +35,20 @@ const OAuth2RedirectHandler = () => {
...
-  }, [profile, logIn, navigate, setProfile, sendRequest, done, setDone]);
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []);
 
   useEffect(() => {
-    if (done) navigate("/");
+    if (done) {
+      const redirect = localStorage.getItem("amazin_redirectListing");
+      localStorage.removeItem("amazin_redirectListing");
+      console.log(redirect + "redirct");
```
Comment (NosBos) [inline frontend/src/pages/OAuth2RedirectHandler/OAuth2RedirectHandler.tsx]: remove

File: frontend/src/components/Listing/AddToCartButton.tsx [hunk 3]
```diff
@@ -29,6 +30,7 @@ export default function AddToCartButton({ listingId }: IAddToCartButtonProps) {
         {},
         `?listingId=${listingId}`
       );
+      console.log(currentCartItem);
```
Comment (NosBos) [inline frontend/src/components/Listing/AddToCartButton.tsx]: remove

**Review by NosBos — APPROVED** [review]
Looks good!!!

File: frontend/src/pages/Listing/Listing.tsx (hunk 1 above)
Comment (kfoxIsProgrammer) [reply frontend/src/pages/Listing/Listing.tsx:47]: Addressed, empty array added

File: frontend/src/pages/OAuth2RedirectHandler/OAuth2RedirectHandler.tsx (hunk 2 above)
Comment (kfoxIsProgrammer) [reply frontend/src/pages/OAuth2RedirectHandler/OAuth2RedirectHandler.tsx]: Removed console.logs

**Pull request merged**